onnx/defs/tensor/defs.cc

```diff
@@ -1638,9 +1641,19 @@ ONNX_OPERATOR_SET_SCHEMA(
               [&](int64_t axis) -> int64_t {
                 return axis < 0 ? axis + input_ndim : axis;
               });
+          // if input consists a symbolic value, return early since shape cannot be inferred
+          for (int i = 0; i < input_ndim; ++i) {
+            if (!input_shape.dim(i).has_dim_value()) {
+              return;
+            }
+          }
 
-          for (int i = 0, j = 0; i < input_ndim; ++i) {
-            if (std::find(axes.begin(), axes.end(), i) != axes.end()) {
+          for (int i = 0; i < input_ndim; ++i) {
+            if (axes_not_specified && input_shape.dim(i).has_dim_value() &&
```

Comment:
nit: ```input_shape.dim(i).has_dim_value()``` can be dropped since it is already checked in previous loop.

Reply:
Good point. I have removed two of them. Thanks